packages/orbit-components/src/Itinerary/context.tsx

```diff
@@ -15,19 +15,21 @@ export const ItineraryProvider = ({
 }: {
   children: React.ReactNode;
 }): React.ReactElement<Context> => {
-  const [widths, setWidths] = React.useState([70]);
-  const [calculatedWidth, setCalculatedWidth] = React.useState(0);
+  const [calculatedWidth, setCalculatedWidth] = React.useState(70);
 
-  React.useEffect(() => {
-    setCalculatedWidth(Math.max(...widths));
-  }, [widths]);
+  const setWidth = React.useCallback(
```

Comment:
To clarify, this `setWidth`, instead of storing all the widths and then have the `useEffect` for `widths` triggered to calculate the max value, it indeed calculates the max value immediately and stores it in the same variable (`calculatedWidth`). With this, we avoid having a state storing an array for just one operation and we get the same desired result.

You confirm?

Reply:
Yes, exactly.
AFAIK, this context is reused for all the SegmentStop components (so there likely are quite a few call to this, meaning lots of unnecessary operations on this array got removed as a result of this change).

We're only interested in the max value.
So doing Math.max(all the possible values) or Math.max(val1, val2) over each call amounts to the same.

> between the width of the date and the minWidth?

Regarding `minWidth`, it's just an initialization thing, so you just call it once with the minWidth, and you're sure that it'll be at least `minWidth`.